source/common/network/apple_dns_impl.cc

```diff
@@ -0,0 +1,325 @@
+#include "common/network/apple_dns_impl.h"
```

Comment:
Sorry, just got around to looking into the details of this PR now. One question I have; what is the threading model here? Who controls the callbacks, which threads do they land on? Maybe a top-level comment explaining this would help. Thanks.

Reply:
https://github.com/envoyproxy/envoy/pull/13074/files#diff-4ebdef3732b2b07e175aba41bcaf62abR280